src/node-jspm-jasmine.js

```diff
@@ -3,52 +3,102 @@ import fs from "fs";
 import jspm from "jspm";
 import path from "path";
 import Jasmine from "jasmine";
+import { Instrumenter, Report } from 'istanbul';
+import { remap, writeReport } from "remap-istanbul"
 
+const coveredModules = []
+const tranpiledFileSuffix = '__node-jspm-jasmine-transpiled__.js'
 export function runTests(opts, errCallback = function() {}) {
 
 	let packagePath = '.';
 
 	try {
-
 		if ( typeof opts.packagePath === 'string' ) {
 			require(path.join(process.cwd(), opts.packagePath, 'package.json'));
 			packagePath = path.join(process.cwd(), opts.packagePath);
 		}
-
 	} catch(ex) {
-
 		let errorMessage = ex.toString();
-
 		if (ex.code === 'MODULE_NOT_FOUND' &&
 			errorMessage.indexOf('package.json') > -1) {
 			errorMessage =
 				'Could not find package.json at custom path: ' +
 				path.join(process.cwd(), opts.packagePath, 'package.json');
-	}
-
+		}
 		errCallback(new Error(errorMessage));
 	}
 
 	jspm.setPackagePath(packagePath);
 
 	const jasmine = new Jasmine();
 	const SystemJS = new jspm.Loader();
+	const Instrument = new Instrumenter();
+	const systemInstantiate = SystemJS.instantiate;
+	const coverageFiles = {}
+	const coverageFilePatterns = opts.coverageFiles || [ 'src/**/*.js' ]
+
+	coverageFilePatterns.forEach( pattern => {
+		glob.sync(path.join(process.cwd(),pattern))
+			.forEach(file => coverageFiles[file] = 1)
+	})
+
+	SystemJS.instantiate = (load) => {
+		const normalizedAddress = load.address.replace( 'file://', '' )
+		if (coverageFiles[normalizedAddress]) {
+			load.address = normalizedAddress;
+			coveredModules.push(load.address);
+			const filename = path.basename(load.name);
+			// Start source map processing
+				// all of this could be avoided if we come to a resolution
+				// on this: https://github.com/SitePen/remap-istanbul/issues/48
+				delete load.metadata.sourceMap.sourcesContent;
```

Comment:
I was just about to reply to your other comment, trying to keep up, lol. So here's what I was saying...

I had some doubts about the load.metadata.sourcemap because jspm only transpiles when it needs to, but then I reasoned that you have to export something if you are to test it, and if you export you must be using ES6, therefore there must always be a source map. With that in mind, notice how you had an issue with a file that has no exports, but also notice how that file should not be getting covered! LOL. Actual test files should not get covered, only app files, thus the option for defining your app files coverageFiles: ["src/shared/**/*.js"]. Instead of accommodating, we should reject these files. Not sure what the best approach is, but maybe we should just let it throw errors for files without sourcemaps, because it likely shouldn't be getting covered anyway and prompt the user to exclude them with the coverageFiles option. But this leaves me with a lingering feeling that maybe there are files that should be covered but don't have source map. I just can't think of any. Can you think of a file that is not a test file, but that can be tested, and that does not export anything therefore would not get transpiled with a source map?

Maybe we can only process files with exports (and thus have sourcemap), and have another option where the user can define exception files (if they even exist)? Or not provide an exception option, wait for a user to complain, and go from there...